RCB-537 dotnet binding

cp2boston opened the pull request.

Commit:
Added source and tests

Commit:
Added comments

- Added required parameters to ctor

Commit:
Added travis

Commit:
Updates found during example testing

Commit:
Docker (may be moved later)

Commit:
Initial example refactor

Commit:
Updated travis

Commit:
Added updated examples

- Updates to source based on feedback from example runs
- Updated README

Commit:
Added Jenkinsfile

- Removed docker files after adding to rapid-development-tools

Commit:
Updated User-Agent

Commit:
Added link to documentation

Commit:
Updated README title

Commit:
Usage added to README

Commit:
Minor code cleanup for consistency

Commit:
Comment update to test Jenkins

Commit:
Timeout bumped to 5 minutes

Commit:
Timespan 5 minutes

- updated unit tests

Commit:
Unused modules

Commit:
Updates per review

- Modified Content to take an object, text, Uri or Filestream
- Cleaned out old modules
- Timeout to 5 minutes

guruxu commented on examples/categories.cs:
```diff
@@ -0,0 +1,44 @@
+        {
+            //To use the C# API, you must provide an API key
+            string apiKey = "Your API key";
+            string altUrl = string.Empty;
```
This looks odd, why not just leave it out?

Commit:
Refactored examples

- Simplified Main to process command line args
- Separate example function to reduce confusion

Commit:
Updated README

guruxu reviewed (approved):
Looking good
